### ISSUE TITLE
Suspending Logging wakes up the car.

### ISSUE THREAD

**Issue by virtualm2000** [adriankumpf/teslamate#111]
Firstly, awesome work!!! Thank you for your time! 

This is the log from teslamate:
20:20:57.421 car_id=1 [info] Suspending logging
20:42:01.535 car_id=1 [info] Suspending logging
Meanwhile the status is "falling asleep for...". 
I can see the car going to sleep by checking when it disconnects from wifi. Few seconds after "Suspending logging" the car comes back online and on wifi. 
Time to try sleeping is set to 21 min.

<img width="351" alt="Screen Shot 2019-08-22 at 21 00 30" src="https://user-images.githubusercontent.com/24257689/63545692-ed747080-c51f-11e9-801b-9ac28c7b919b.png">

This is the log from my DNS server. The connection to Tesla servers is visible.

2019-08-22 20:20:52 | A | owner-api.teslamotors.com | teslamate.local | OK (forwarded) | CNAME
2019-08-22 20:41:57 | A | owner-api.teslamotors.com | teslamate.local | OK (forwarded) | CNAME

Is this a design feature? One workaround is to increase the Time to try sleeping to 60 min. I have a script to resume logging anyway when connect to the car by BT.

Best regards,
Mihai

**Comment by adriankumpf** [comment]
Thanks for the detailed description. That helps a lot.

Apparently not all cars can handle the default value of 21 minutes. Some models fall asleep after 12 minutes even, others need more time than that.

This is how it works: during the  "Time to try sleeping" of 21 minutes Teslamate does not poll the car **at all**. As long as there are no external interference factors like Android widgets, other loggers or nearby Bluetooth devices etc. nothing prevents the car from falling asleep. Afterwards, Teslamate sends a request to the Tesla `vehicles` API (which does not require the car to be online) to request the state of the car. If the state is reported as _asleep_ or _offline_, Teslamate continues to poll the `vehicles` API twice a minute to figure out its current state.

If, however, the car is reported as _online_, another request to the `vehicle data` API is made to figure out whether it is driving or charging, for example. Requests to the `vehicle data` API potentially prevent the car from falling asleep. But as mentioned earlier, this request is only triggered if the car did not fall asleep and is instead still _online_. If no driving or charging was detected, Teslamate repeats the "Time to try sleeping" phase.

So, to summarize: 
* After the configured "Time to try sleeping" Teslamte will continue to poll the car. It depends on the reported state of the car which API endpoint is being called.
* Teslamate does not prevent the car from sleeping / does not wake up the car.
* However, some cars need more than 21 minutes to fall asleep. Once the car is reported as asleep, Teslamate will notice and will not wake it up.

Your workaround is a perfectly fine solution here especially if you use Tasker or the like to resume logging when necessary.

I'd like to know how far you can reduce the "Time to try sleeping" value ...

**Comment by virtualm2000** [comment]
The car apparently goes to sleep just fine after approximately 12 minutes. I can see that by checking my wifi network.

I ran the api for vehicles myself when the car was sleeping and the server reported the car as online:

 Connection #0 to host owner-api.teslamotors.com left intact
{"response":[{"id":xxxxxxxxxxxxxxxx,"vehicle_id":xxxxxxxxxxxx,"vin":"5YJXDCExxxxxxxxxx","display_name":"","option_codes":"blahxxxxx","color":null,"tokens":["xxxxxxxxxxxxxxx","xxxxxxxxxxxxx"],"state":"online","in_service":false,"id_s":"xxxxxxxxxxxxxxxxxx","calendar_enabled":true,"api_version":6,"backseat_token":null,"backseat_token_updated_at":null}],"count":1

That explains the second api call to vehicle data from teslamate which wakes up the car.

My car has the MCU1 which has the old energy management system. In my energy saving settings I have the Always connected checked. And I think that is the reason why the car is reported as online even when it's sleeping.

Tomorrow I'll disable always connected and see if I get offline in the status field.

**Comment by adriankumpf** [comment]
Ah, I see. Interesting. According to [this article](https://support.teslafi.com/knowledge-bases/2/articles/161-my-vehicle-is-not-sleeping) 'Energy Savings' needs to be turned on as well.

**Comment by virtualm2000** [comment]
Disabling Always connected solved the problem.

08:50:37.917 car_id=1 [info] Start / :asleep

Thank you!
